src/zenml/models/pipeline_build_models.py

```diff
@@ -34,10 +34,111 @@
+    """Pipeline build item.
 
-    configuration: PipelineBuild
+    Attributes:
+        image: The image name or digest.
+        settings_checksum: Checksum of the settings used for the build.
+    """
+
+    image: str = Field(title="The image name or digest.")
+    settings_checksum: Optional[str] = Field(
+        title="The checksum of the build settings."
+    )
+
+
+class PipelineBuildBaseModel(pydantic_utils.YAMLSerializationMixin):
+    """Base model for pipeline builds.
+
+    Attributes:
+        images: Docker images of this build.
+        is_local: Whether the images are stored locally or in a container
+            registry.
+    """
+
+    images: Dict[str, BuildItem] = Field(
+        default={}, title="The images of this build."
+    )
+    is_local: bool = Field(
+        title="Whether the build images are stored in a container registry or locally.",
+    )
+
+    @staticmethod
+    def get_key(key: str, step: Optional[str] = None) -> str:
+        """Get the image key.
+
+        Args:
+            key: The image key.
+            step: The name of the step for which the image was built.
+
+        Returns:
+            The image key.
+        """
+        if step:
+            return f"{step}.{key}"
+        else:
+            return key
+
+    def get_image(self, key: str, step: Optional[str] = None) -> str:
+        """Get the image built for a specific key.
+
+        Args:
+            key: The key for which to get the image.
+            step: The name of the step for which to get the image. If no image
+                exists for this step, will fallback to the pipeline image for
+                the same key.
+
+        Returns:
+            The image name or digest.
+        """
+        return self._get_item(key=key, step=step).image
+
+    def get_settings_checksum(
+        self, key: str, step: Optional[str] = None
+    ) -> Optional[str]:
+        """Get the settings checksum for a specific key.
+
+        Args:
+            key: The key for which to get the checksum.
+            step: The name of the step for which to get the checksum. If no
+                image exists for this step, will fallback to the pipeline image
+                for the same key.
+
+        Returns:
+            The settings checksum.
+        """
+        return self._get_item(key=key, step=step).settings_checksum
+
+    def _get_item(self, key: str, step: Optional[str] = None) -> BuildItem:
+        """Get the item for a specific key.
+
+        Args:
+            key: The key for which to get the item.
+            step: The name of the step for which to get the item. If no item
+                exists for this step, will fallback to the item for
+                the same key.
+
+        Raises:
+            KeyError: If no item exists for the given key.
+
+        Returns:
+            The build item.
+        """
+        if step:
+            try:
+                combined_key = self.get_key(key=key, step=step)
```

Comment:
This seems to be the only place where the function above is used, in that case I think we could get rid of it

```suggestion
                combined_key = return f"{step}.{key}"
```

Reply:
Actually also used in a different place, so I think it needs to exist, I'll rename it though